liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -649,98 +648,89 @@ public void rollback(int changesToRollback, String rollbackScript, String contex
+                //
+                Connection connection = getConnection(changeLog);
+                if (connection != null) {
+                    rollbackOperation = hubUpdater.preUpdateHub("ROLLBACK", operationCommand, connection, changeLogFile, contexts, labelExpression, listLogIterator);
+                }
 
-                    //
-                    // If we are doing Hub then set up a HubChangeExecListener
-                    //
-                    if (connection != null) {
-                        changeExecListener = new HubChangeExecListener(rollbackOperation, changeExecListener);
-                    }
+                //
+                // If we are doing Hub then set up a HubChangeExecListener
+                //
+                if (connection != null) {
+                    changeExecListener = new HubChangeExecListener(rollbackOperation, changeExecListener);
+                }
 
-                    //
-                    // Create another iterator to run
-                    //
-                    ChangeLogIterator logIterator = new ChangeLogIterator(database.getRanChangeSetList(), changeLog,
-                            new AlreadyRanChangeSetFilter(database.getRanChangeSetList()),
-                            new ContextChangeSetFilter(contexts),
-                            new LabelChangeSetFilter(labelExpression),
-                            new DbmsChangeSetFilter(database),
-                            new IgnoreChangeSetFilter(),
-                            new CountChangeSetFilter(changesToRollback));
+                //
+                // Create another iterator to run
+                //
+                ChangeLogIterator logIterator = new ChangeLogIterator(database.getRanChangeSetList(), changeLog,
+                        new AlreadyRanChangeSetFilter(database.getRanChangeSetList()),
+                        new ContextChangeSetFilter(contexts),
+                        new LabelChangeSetFilter(labelExpression),
+                        new DbmsChangeSetFilter(database),
+                        new IgnoreChangeSetFilter(),
+                        new CountChangeSetFilter(changesToRollback));
 
-                    doRollback(bufferLog, rollbackScript, logIterator, contexts, labelExpression, hubUpdater, rollbackOperation);
-                }
-                catch (Throwable t) {
-                    handleRollbackException(t, hubUpdater, rollbackOperation, bufferLog, operationCommand);
-                    throw t;
-                } finally {
-                    handleRollbackFinally(lockService);
-                }
+                doRollback(bufferLog, rollbackScript, logIterator, contexts, labelExpression, hubUpdater, rollbackOperation);
+            }
+            catch (Throwable t) {
+                handleRollbackException(t, hubUpdater, rollbackOperation, bufferLog, operationCommand);
+                throw t;
+            } finally {
+                handleRollbackFinally(lockService);
```

Comment:
Shouldn't all this code be replaced with a call to the rollback-count command step? Or is that not part of this story?